lib/spawn/cluster/state_handoff/controllers/nats_kv_controller.ex

```diff
@@ -0,0 +1,144 @@
+defmodule Spawn.Cluster.StateHandoff.Controllers.NatsKvController do
+  @moduledoc """
+  This handles state handoff in a cluster.
+
+  It uses the Nats jetstream library to handle a distributed state, which is an eventually consistent replicated data type.
+  """
+  require Logger
+
+  alias Actors.Config.PersistentTermConfig, as: Config
+
+  import Spawn.Utils.Common, only: [generate_key: 1, actor_host_hash: 0]
+
+  @behaviour Spawn.Cluster.StateHandoff.ControllerBehaviour
+
+  @type node_type :: term()
+
+  @type opts :: Keyword.t()
+
+  @type data :: any()
+
+  @type new_data :: data()
+
+  @type id :: Eigr.Functions.Protocol.Actors.ActorId.t()
+
+  @type host :: Actors.Registry.HostActor.t()
+
+  @type hosts :: list(Actors.Registry.HostActor.t())
+
+  @type timer :: {atom(), integer()}
+
+  @doc """
+  Cluster HostActor cleanup
+  """
+  @impl true
+  def clean(node, %{} = data) do
+    Logger.debug("Received cleanup action from Node #{inspect(node)}")
+
+    system = Config.get(:actor_system_name)
+
+    # hard fail in case of error
+    :ok = Jetstream.API.KV.purge_key(conn(), bucket_name(), "#{system}.#{actor_host_hash()}.*")
```

Comment:
Here you purge all hosts! RIght? Because you don´t pass node

Reply:
Ok! But I think it's worth a comment in the code to remember this in the future.